XMonad/Util/Stack.hs

```diff
@@ -123,6 +124,18 @@ toTags Nothing = []
 toTags (Just s) = map Left (reverse . W.up $ s) ++ [Right . W.focus $ s]
                   ++ map Left (W.down s)
 
+-- | @differentiate zs xs@ takes the first @z@ from @z2 that also belongs to
+-- @xs@ and turns @xs@ into a stack with @z@ being the current element. Acts
+-- as 'XMonad.StackSet.differentiate' if @zs@ and @xs@ don't intersect.
+differentiateTwo :: Eq q => [q] -> [q] -> Zipper q
```

Comment:
Does the name "differentiateTwo" make any sense? It seems the naming of the integrate/differentiate pair of functions in XMonad.StackSet is explained by this: https://en.wikibooks.org/wiki/Haskell/Zippers#Differentiation_of_data_types, but "differentiateTwo" doesn't fit into that scheme (but then I don't really understand this stuff, and probably never will). It's really more of a "find first z if any and make a zipper focused there" I think. Not sure how to name it though.

Reply:
I was wondering that also, but I can't think of a better name either 😀

Reply:
> Does the name "differentiateTwo" make any sense?

Not in the actual definitional sense of "differentiate", no—this is not me differentiating `[a] × [a]` or something. This is just me being bad at names. I'm very open for suggestions here, so if you think of something let me know

Reply:
`differentiateAt`
`zipperFocusedAt`
`zipperFocusedAtFirst` (this one is probably closest to the actual behaviour, but also the longest)